### DISCUSSION TITLE
Is aiMeshMorphAnim::mName useless? How can I found the mesh to be animated then?

### DISCUSSION THREAD

**Discussion by Lucipetus** [assimp/assimp#4144] [Q&A] [answered]
Here's what the doc says about aiMeshMorphAnim::mName:
/** Name of the mesh to be animated. An empty string is not allowed,
     *  animated meshes need to be named (not necessarily uniquely,
     *  the name can basically serve as wildcard to select a group
     *  of meshes with similar animation setup)*/

But I get something like "Cube*0". The "Cube" appears to be the name of the FBX Model.

I named the base mesh "cube_mesh" in blender, as a result the loaded aiMesh is with the name "cube_mesh"! The target meshes were also given meaningful names.

So "Cube*0" is NOT a name of any of the meshes!!!

"0" seems to be the index of the FBX Geometry in a FBX Model. The thing is, the loaded aiScene does NOT contain any information of the FBX Model, as far as I can find! There's no way for me to find the mesh with "Cube*0".

// FBXConverter.cpp
```
                    if (geo) {
                        std::vector<const Connection *> geoConnections = doc.GetConnectionsBySourceSequenced(geo->ID(), "Model");
                        for (const Connection *geoConnection : geoConnections) {
                            auto model = dynamic_cast<const Model *>(geoConnection->DestinationObject());
                            if (model) {
                                auto geoIt = std::find(model->GetGeometry().begin(), model->GetGeometry().end(), geo);
                                auto geoIndex = static_cast<unsigned int>(std::distance(model->GetGeometry().begin(), geoIt));
                                auto name = aiString(FixNodeName(model->Name() + "*"));
```

-----------------------------------------------------------------------------------------------------------

aiMesh gets the name of the FBX Geometry if it's not empty, while aiMeshMorphAnim gets the name of the FBX Model, with the index of the geometry appended.

This is broken.

// FBXConverter::SetupEmptyMesh
```
    // set name
    std::string name = mesh.Name();
    if (name.substr(0, 10) == "Geometry::") {
        name = name.substr(10);
    }

    if (name.length()) {
        out_mesh->mName.Set(name);
    } else {
        out_mesh->mName = parent->mName;
    }
```
------------------------------------
I've raised an issue: https://github.com/assimp/assimp/issues/4149
I still have no idea about the "wildcard" part. Can someone explain?

------------------------------------
I should probably have seen this earlier.

```
/** The number of attachment meshes. Note! Currently only works with Collada loader. */
    unsigned int mNumAnimMeshes;

    /** Attachment meshes for this mesh, for vertex-based animation.
     *  Attachment meshes carry replacement data for some of the
     *  mesh'es vertex components (usually positions, normals).
     *  Note! Currently only works with Collada loader.*/
    C_STRUCT aiAnimMesh **mAnimMeshes;
```

So if mAnimMeshes isn't officially working with FBX, there's little wonder aiMeshMorphAnim::mName is broken.

**Comment by kimkulling** [comment, marked as answer]
Thanks for your update!
